keras_nlp/models/bart/bart_seq_2_seq_lm_preprocessor.py

```diff
@@ -185,3 +187,104 @@ def call(self, x, y=None, sample_weight=None):
         y = decoder_token_ids[..., 1:]
         sample_weight = decoder_padding_mask[..., 1:]
         return pack_x_y_sample_weight(x, y, sample_weight)
+
+    def generate_preprocess(
+        self,
+        x,
+        sequence_length=None,
+    ):
+        """Convert encoder and decoder input strings to integer token inputs for generation.
+
+        Similar to calling the layer for training, this method takes in a dict
+        containing `"encoder_text"` and `"decoder_text"`, with strings or tensor
+        strings for values, tokenizes and packs the input, and computes a padding
+        mask masking all inputs not filled in with a padded value.
+
+        For the encoder inputs, the above step is exactly the same as for
+        training. For the decoder inputs, this method does not compute labels
+        and will never append a `tokenizer.end_token_id` to the end of the
+        sequence (as generation is expected to continue at the end of the
+        inputted prompt).
+        """
+        # If `sequence_length` is not provided, we use the default value.
+        # We decrement this value by one since we need to add `end_token_id` to
+        # the *beginning* of the decoder sequence.
+        sequence_length = (
+            sequence_length - 1
+            if sequence_length
+            else self._decoder_sequence_length - 1
+        )
+
+        if isinstance(x, dict):
+            encoder_text = x["encoder_text"]
+            decoder_text = x["decoder_text"]
+        else:
+            encoder_text = x
+            decoder_text = tf.fill((tf.shape(encoder_text)[0],), "")
+
+        # Tokenize the encoder inputs. We can use the preprocessor directly
+        # here. At the same time, the preprocessor will also do pre-checks on
+        # the inputs.
+        preprocessed_inputs = self(
```

Comment:
This feels awkward to route through `call`, and discard the labels. Why not just show the full flow with packer layers and tokenizers here, similar to [for gpt2](https://github.com/keras-team/keras-nlp/blob/7c45b9612c65663cdddd01ad79087ff9cdedaa23/keras_nlp/models/gpt2/gpt2_causal_lm_preprocessor.py#L146-L154). This will also help anyone wanting to copy paste the logic here to tweak it.